### PR TITLE
Core: Fix - Vatom view reuse pool

### PR DESCRIPTION
**Pull request by cjmconie**
Fixes and issue where a load closure could affect the state of a vatom-view even after the vatom-view's vatom had changed..

### CODE REVIEW

File: BlockV/Face/Vatom View/VatomView.swift [hunk 1]
```diff
@@ -361,49 +362,82 @@ open class VatomView: UIView {
...
-        let newSelectedFaceView: FaceView = viewType.init(vatom: vatom,
-                                                          faceModel: newSelectedFaceModel)
-        newSelectedFaceView.delegate = self
+//            printBV(info: "Face view for face model: \(faceViewType)")
 
-        // replace currently selected face view with newly selected
-        self.replaceFaceView(with: newSelectedFaceView)
-        // inform delegate
-        self.vatomViewDelegate?.vatomView(self, didSelectFaceView: .success(newSelectedFaceView))
+            //let selectedFaceView: FaceView = ImageFaceView(vatom: vatom, faceModel: selectedFace, host: self)
```
Comment (MalcolmnDEV) [inline BlockV/Face/Vatom View/VatomView.swift]: remove comment?

File: BlockV/Face/Vatom View/VatomView.swift [hunk 2]
```diff
@@ -361,49 +362,82 @@ open class VatomView: UIView {
...
-        
-        //printBV(info: "Face view for face model: \(faceViewType)")
+                return
+            }
 
-        //let selectedFaceView: FaceView = ImageFaceView(vatom: vatom, faceModel: selectedFace, host: self)
-        let newSelectedFaceView: FaceView = viewType.init(vatom: vatom,
-                                                          faceModel: newSelectedFaceModel)
-        newSelectedFaceView.delegate = self
+//            printBV(info: "Face view for face model: \(faceViewType)")
```
Comment (MalcolmnDEV) [inline BlockV/Face/Vatom View/VatomView.swift]: remove print ?

File: BlockV/Face/Vatom View/VatomView.swift [hunk 3]
```diff
@@ -361,49 +362,82 @@ open class VatomView: UIView {
...
+//        printBV(info: "Face model changed - Replacing face view.")
+
+        // 2. check if the face model has not changed
 
-        // replace currently selected face model
-        self.selectedFaceModel = newSelectedFaceModel
+        if (vatom.props.templateVariationID == oldVatom?.props.templateVariationID) &&
+            (newSelectedFaceModel == self.selectedFaceModel) {
+
+            //printBV(info: "Face model unchanged - Updating face view.")
```
Comment (MalcolmnDEV) [inline BlockV/Face/Vatom View/VatomView.swift]: remove print?

File: BlockV/Face/Vatom View/VatomView.swift [hunk 4]
```diff
@@ -361,49 +362,82 @@ open class VatomView: UIView {
...
+         Here we check:
+         
+         A. If selected face model is still equal to the current. This is necessary since the face may
+         change as a result of the publisher modifying the face (typically via a delete/add operation).
+         
+         B. If the new vatom and the previous vatom share a template variation. This is needed since resources are
+         defined at the template variation level.
+         */
+
+//        printBV(info: "Face model changed - Replacing face view.")
```
Comment (MalcolmnDEV) [inline BlockV/Face/Vatom View/VatomView.swift]: remove print

File: BlockV/Face/Vatom View/VatomView.swift [hunk 5]
```diff
@@ -361,49 +362,82 @@ open class VatomView: UIView {
             //printBV(error: "Face Selection Procedure (FSP) returned without selecting a face model.")
```
Comment (MalcolmnDEV) [inline BlockV/Face/Vatom View/VatomView.swift]: remove print?

File: BlockV/Face/Vatom View/VatomView.swift (hunk 1 above)
Comment (cjmconie) [reply BlockV/Face/Vatom View/VatomView.swift]: Removed

**Pull request merged**